rasa/core/featurizers.py

```diff
@@ -563,7 +563,7 @@ def __init__(
     ) -> None:
 
         super().__init__(state_featurizer, use_intent_probabilities)
-        self.max_history = max_history
+        self.max_history = max_history or self.MAX_HISTORY_DEFAULT
```

Comment:
this will not allow set it to `None`, which we need for `RulePolicy`

Reply:
We explicitly set it to `None` in the `RulePolicy` by passing in the initialized featurizer (as opposed to `None` as before)

Reply:
hm, I think resetting to None is less intuitive. But it's not super important, e2e will change that

Reply:
in e2e it is None by default

Reply:
Ok 👍 I basically just changed it back to the `master` version and then did the "hack" in the RulePolicy to pass in `None`. Will leave it like that then for now.